From 6c39859ddcc94b311c653dee42d32d2412a6f406 Mon Sep 17 00:00:00 2001
From: Martin Pitt <martinpitt@gnome.org>
Date: Wed, 27 Jun 2012 09:25:37 +0200
Subject: [PATCH] gio/tests/contenttype: Call g_content_type_guess() with
 valid data len

g_content_type_guess() requires specifying a valid data length. Fixes a
segfault when running the test.

Bug: https://bugzilla.gnome.org/show_bug.cgi?id=674452
---
 gio/tests/contenttype.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/gio/tests/contenttype.c b/gio/tests/contenttype.c
index 7188894..9e0f9af 100644
--- a/gio/tests/contenttype.c
+++ b/gio/tests/contenttype.c
@@ -55,14 +55,14 @@ test_guess (void)
   g_free (res);
   g_free (expected);
 
-  res = g_content_type_guess ("test.pot", (guchar *)"ABC abc", -1, &uncertain);
+  res = g_content_type_guess ("test.pot", (guchar *)"ABC abc", 7, &uncertain);
   expected = g_content_type_from_mime_type ("application/vnd.ms-powerpoint");
   g_assert (g_content_type_equals (expected, res));
   g_assert (uncertain);
   g_free (res);
   g_free (expected);
 
-  res = g_content_type_guess ("test.otf", (guchar *)"OTTO", -1, &uncertain);
+  res = g_content_type_guess ("test.otf", (guchar *)"OTTO", 4, &uncertain);
   expected = g_content_type_from_mime_type ("application/x-font-otf");
   g_assert (g_content_type_equals (expected, res));
   g_assert (!uncertain);
-- 
1.7.10.4

